### ISSUE TITLE
uname: Redox release version is outdated

### ISSUE THREAD

**Issue by michaelbachner** [redox-os/redox#1056]
**Reproduction**: Type uname -r.

**Expected behavior**: It prints out 0.3.2 in Release 0.3.2.

**Actual behavior**: 
![image](https://user-images.githubusercontent.com/18561536/30252210-4be6515c-966e-11e7-867f-6d6020aa96c3.png)

**Redox release**: 0.3.2

**Misc**: I run the redox_0.3.2.iso

**Comment by ids1024** [comment]
The version comes from the kernel's `Cargo.toml`. Kernel versions aren't necessarily the same as Redox versions (but probably should be, without any reason otherwise). So I guess the kernel should be version bumped before each Redox release?

**Comment by michaelbachner** [comment]
You mean this line, right? https://github.com/redox-os/kernel/blob/cf45bd34e107df2faf590475546a7d7ac34114db/Cargo.toml#L3 

As a user I expect release to be the OS release. If it's meant to be the kernel version, then uname should have another parameter for outputting the OS release.

In my opinion kernel and os version should be in sync.

**Comment by ids1024** [comment]
Yes, that line.

[POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/uname.html) does seem to specify that `uname -r` prints the OS version; the Linux man page specifies that it returns the kernel version. The two should probably be the same for something like Redox (unlike Linux, where the kernel and distribution are developed independently).

The current implementation uses a `sys:uname` psuedo-file provided by the kernel, so that would have to be changed to use a different version; such as a file containing the OS version. But I think it's best to keep the current method but bump the kernel version.

**Comment by jackpot51** [comment]
The behavior is correct. `uname -r` is the *kernel* version. I recommend instead creating an `/etc/os-release` file, similar to the one in Linux:

Here is one from Ubuntu:
```
NAME="Ubuntu"
VERSION="17.10 (Artful Aardvark)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu Artful Aardvark (development branch)"
VERSION_ID="17.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=artful
UBUNTU_CODENAME=artful
```

**Comment by michaelbachner** [comment]
Is that really necessary? If OS and kernel are developed together, why should they have different version numbers? Is there a good reason for the os-release file?

**Comment by jackpot51** [comment]
If the kernel is stable, and does not change, why should the version change when the OS version does?

Also, why should a kernel change trigger an entire OS version change?

**Comment by ids1024** [comment]
Fair enough. And if Redox is intended to be rolling release, there aren't real releases, but semi-arbitrary snapshots. Perhaps the versions should be dated, like Archlinux (the current Arch ISO is 2017.09.01).

(Redox isn't entirely rolling release yet due to the state of the package manager, and the fact that the kernel and bootloader aren't managed through the package manager.)
